ls1intum/Artemis
# [Feature] Gitlab/Jenkins Pseudonyms

### Checklist
- [x] @ivanchimeno has added some test cases 

### Motivation and Context
GitLab and Jenkins allow any user to see any user's basic information. This leads to the problem that any student / user can map a pseudonym (login name) to the actual name of the user. This is a privacy issue. See also https://gitlab.com/gitlab-org/gitlab/-/issues/14123 

#### Jenkins
![image](https://user-images.githubusercontent.com/22198798/112122812-f11b1e00-8bc0-11eb-83d6-89709b2820a4.png)

#### GitLab
![image](https://user-images.githubusercontent.com/22198798/112123230-5e2eb380-8bc1-11eb-9514-e97acacfb80a.png)



### Description
I've added two new properties (this should be the most flexible way):
* gitlab.use-pseudonyms: false
* jenkins.use-pseudonyms: false

Enabling the resp. property forces artemis to send "User \<loginname\>" as user's name to gitlab or jenkins.

### Steps for Testing


1. Enable the properties (or only one iff you only want to test one system)
2. Login & create a new user
3. Check whether the new user's name is visible in jenkins resp. gitlab

### Test Coverage
I cannot see this.


If you have any questions, feel free to ask :)
